geoutils/pointcloud/pointcloud.py

```diff
@@ -1 +1,393 @@
+
+    @property
+    def ds(self) -> gpd.GeoDataFrame:
+        """Geodataframe of the point cloud."""
+        # We need to override the Vector method to introduce the is_loaded dynamic for LAS files
+        if not self.is_loaded:
+            self.load()
+        return self._ds  # type: ignore
+
+    @ds.setter
+    def ds(self, new_ds: gpd.GeoDataFrame | gpd.GeoSeries) -> None:
+        """Set a new geodataframe for the point cloud."""
+        # We need to override the setter Vector method because we have overridden the property method
+        # (even if the code below is the same)
+        if isinstance(new_ds, gpd.GeoDataFrame):
+            self._ds = new_ds
+        elif isinstance(new_ds, gpd.GeoSeries):
+            self._ds = gpd.GeoDataFrame(geometry=new_ds)
+        else:
+            raise ValueError("The dataset of a vector must be set with a GeoSeries or a GeoDataFrame.")
+
+    @property
+    def crs(self) -> CRS:
+        """Coordinate reference system of the vector."""
+        # Overriding method in Vector
+        if self.is_loaded:
+            return super().crs
+        else:
+            return self._crs
+
+    @property
+    def bounds(self) -> BoundingBox:
+        # Overriding method in Vector
+        if self.is_loaded:
+            return super().bounds
+        else:
+            return self._bounds
+
+    #####################################
+    # NEW METHODS SPECIFIC TO POINT CLOUD
+    #####################################
+
+    @property
+    def data(self) -> NDArrayNum:
+        """
+        Data of the point cloud.
+
+        Points to the data column of the geodataframe, equivalent to calling self.ds[self.data_column].
+        """
+        # Triggers the loading mechanism through self.ds
+        return self.ds[self.data_column]
+
+    @data.setter
+    def data(self, new_data: NDArrayNum) -> None:
+        """Set new data for the point cloud."""
+
+        self.ds[self.data_column] = new_data
+
+    @property
+    def all_columns(self) -> pd.Index:
+        """Index of all columns of the point cloud, excluding the column of 2D point geometries."""
+        # Overriding method in Vector
+        if self.is_loaded:
+            all_columns = super().columns
+            all_columns_nongeom = all_columns[all_columns != "geometry"]
+            return all_columns_nongeom
+        else:
+            return self._all_columns
+
+    @property
+    def data_column(self) -> str:
+        """Name of data column of the point cloud."""
+        return self._data_column
+
+    @data_column.setter
+    def data_column(self, new_data_column: str) -> None:
+        self.set_data_column(new_data_column=new_data_column)
+
+    def set_data_column(self, new_data_column: str) -> None:
+        """Set new column as point cloud data column."""
+
+        if new_data_column not in self.all_columns:
+            raise ValueError(
+                f"Data column {new_data_column} not found among columns, available columns "
```

Comment:
Maybe add a break. "not found among columns, available columns..." -> "not found among columns. Available columns..."